python/metatensor-torch/metatensor/torch/atomistic/ase_calculator.py

```diff
@@ -352,18 +368,54 @@ def calculate(
                 energy.backward(-torch.ones_like(energy))
 
         with record_function("ASECalculator::convert_outputs"):
-            if "forces" in properties:
+            if "forces" in properties_to_calculate:
                 forces_values = system.positions.grad.reshape(-1, 3)
                 forces_values = forces_values.to(device="cpu").to(dtype=torch.float64)
                 self.results["forces"] = forces_values.numpy()
 
-            if "stress" in properties:
+            if "stress" in properties_to_calculate:
                 stress_values = -strain.grad.reshape(3, 3) / atoms.cell.volume
                 stress_values = stress_values.to(device="cpu").to(dtype=torch.float64)
                 self.results["stress"] = _full_3x3_to_voigt_6_stress(
                     stress_values.numpy()
                 )
 
+    def request_properties_every_n_steps(
```

Comment:
I still don't think this should be part of the same object. The calculator should follow the ASE API for calculators, and we should provide a separate tool (class, function, …) to handle calculations that have to run at other times.

Reply:
Ok, I can extract it into a standalone function